### ISSUE TITLE
Testkit support for asserting all different effects

### ISSUE THREAD

**Issue by johanandren** [lightbend/akkaserverless-java-sdk#322]
Currently the Result type returned expects a reply message, we should support capturing the other types of effects and asserting those as well.

**Comment by johanandren** [comment]
Merged, but one thing that is still missing is some way to inspect/assert side effects (I think we can add that later).

**Comment by patriknw** [comment]
@johanandren @franciscolopezsancho this ticket is still open. Do we have more work planned for this ticket or can we close it?

**Comment by franciscolopezsancho** [comment]
It felt out of my radar, sorry. We need it for Event Sourced Entities and Value Entities, right?

**Comment by johanandren** [comment]
Side effect access still missing in ValueEntityResult and EventSourcedResult yes.

Replicated entities we don't have any testkit for (yet) but that should be tracked in a different separate issue.

**Comment by franciscolopezsancho** [comment]
this is blocked by #722

**Comment by franciscolopezsancho** [comment]
it's done now for Event Sourced Entities and Actions. Still missing Value entities.
